### PR TITLE
Update chaos-mesh/api dependencies

### PR DESCRIPTION
**Pull request by fasmat**
### What problem does this PR solve?

Updates dependencies of `chaos-mesh/api`. `sigs.k8s.io/controller-runtime` had breaking changes between v0.13.1 and v0.14.0 that required a small change to the tests to make the update work.

### Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`
- Need to **cheery-pick to release branches**
  - [ ] release-2.5
  - [ ] release-2.4

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] **Breaking backward compatibility**

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by fasmat** [comment]
Sorry I wanted to CC not assign 🙁

**Comment by STRRL** [comment]
Hi @fasmat , thanks for your contribution!

Please execute `go mod tidy` and fix the noted issues by that command. Then execute `make check` for executing linters.

And please append a brief description in `CHANGELOG.md` with this PR and included changes.

**Comment by STRRL** [comment]
> I want to label this PR with "no-need-update-changelog" since I don't find it noteworthy to be included in the CHANGELOG but I can't add labels to the PR.

it should be listed in the "Changed" section, because of the upgrade of library.

**Comment by fasmat** [comment]
@STRRL I added my changes to the CHANGELOG.md file and ran `make check`.

For the checks to successfully pass I had to update the other modules in the repository as well. It might be worth considering using [go workspaces](https://go.dev/doc/tutorial/workspaces) in the future. They make it easier for all modules to align in their dependencies and make `replace` directives in the `go.mod` files for modules of the same repository unnecessary (and `replace` directives in the `go.work` file are applied to all modules).

There were some updated `.yaml` and `.json` files after running `make check`. I assume these changes were missing from an earlier PR and since it looks like documentation to me I committed them in this PR as well.

Let me know when something is still missing 🙂

**Comment by STRRL** [comment]
> It might be worth considering using [go workspaces](https://go.dev/doc/tutorial/workspaces) in the future.

Sounds great! Using go workspace would make the libraries' version much easier to align.

> There were some updated .yaml and .json files after running make check. I assume these changes were missing from an earlier PR and since it looks like documentation to me I committed them in this PR as well.

I took a look at changes on yaml and json file. The upstream changes on description/comment cause these change, introduced by the kubernetes upgrading 0.24.x -> 0.26.x, that's fine. :heart:

**Comment by STRRL** [comment]
Hi @fasmat , it is almost accomplished! Thanks!

It seems e2e test failed to compile because the version of ginkgo are misconfigured, we might use ginkgo `2.4.0` which used on kubernetes 1.26.1 instead of `2.7.0 // indirectly` .

Try to align the version and execute `make e2e` to compile the e2e test codes.

> `make e2e` would also execute the e2e tests suites, but it would fail if we do not setup the test environment. So just let it go, I think we might only need to focus on the compilation errors. 

Also, there are some API changes of kubernetes e2e test framework between 1.24.x and 1.26.x. These changes could also be unraveled when you execute `make e2e`.

**Comment by fasmat** [comment]
I updated the incompatible dependency; `ginkgo` fixed the break from API compatibility with version `v2.7.1`.

This required me to update a few additional dependencies, but both `make check` and `make e2e` pass now on my machine.

**Comment by STRRL** [comment]
It looks like we should also bump ginkgo in `.go` file to v2..

It might be lot of tricky, I would take a try 🤔

File: e2e-test/go.mod [hunk 1]
```diff
@@ -1,169 +1,134 @@
...
-	k8s.io/kubectl => k8s.io/kubectl v0.24.5
-	k8s.io/kubelet => k8s.io/kubelet v0.24.5
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.24.5
-	k8s.io/metrics => k8s.io/metrics v0.24.5
-	k8s.io/mount-utils => k8s.io/mount-utils v0.24.5
-	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.24.5
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.24.5
-	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.24.5
-	k8s.io/sample-controller => k8s.io/sample-controller v0.24.5
-	vbom.ml/util => github.com/fvbommel/util v0.0.2
```
Comment (STRRL) [inline e2e-test/go.mod:166]: it seems the issue about e2e is caused by these changes.

The e2e test part import the kubernetes e2e test framework, we need manually pin the kubernetes version.

We have a script to do that: `hack/update-kubernetes-library.sh`, that might be trick.

To fix that, I would push commits to this PR directly. Wish you do not mind this.

Comment (fasmat) [reply e2e-test/go.mod:166]: Sure go ahead 😊

Comment (mkfdoherty) [reply e2e-test/go.mod:166]: I thought I would jump in to see if there is potentially a misunderstanding here.

@STRRL are you suggesting that @fasmat run

```bash
# The command will adjust ./go.mod and ./go.sum in the base directory of the repo
./hack/update-kubernetes-library.sh v1.26.1
```

and that he should commit and push those changes to this PR?

Are there any additional changes that need to be made to Ginkgo versioning for this PR to be ready? Will this require code changes?

Comment (fasmat) [reply e2e-test/go.mod:166]: I re-added the replacements to the `go.mod` file and updated `ginkgo` to `v2.8.0`.

**Comment by fasmat** [comment]
@STRRL @mkfdoherty : I fixed the broken tests. `make test` should now pass without errors.

`ginkgo` had some changes between v1 and v2 regarding async tests. I only did the minimum necessary to get those tests running again, but it might be worth in the future to take a look at them and adapt them to the new async model.

File: e2e-test/go.mod (hunk 1 above)
Comment (STRRL) [reply e2e-test/go.mod:166]: > I thought I would jump in to see if there is potentially a misunderstanding here.
> 
> @STRRL are you suggesting that @fasmat run
> 
> ```shell
> # The command will adjust ./go.mod and ./go.sum in the base directory of the repo
> ./hack/update-kubernetes-library.sh v1.26.1
> ```
> 
> and that he should commit and push those changes to this PR?
> 
> Are there any additional changes that need to be made to Ginkgo versioning for this PR to be ready? Will this require code changes?

Sorry for the late response.

**Comment by STRRL** [comment]
> For the remaining test that fail I'm not sure they are related to my changes thinking or at least I'm not sure how to fix them.
> 
> The test coverage report upload fails with an HTTP 404 error. Some E2E tests fail because of missing apt packages, others because the tests time out. How to proceed?

It is caused by using ginkgo v1 and v2 simultaneously.  And the e2e tests suite failed to execute. I am going to try to purge the ginkgo v1.

**Comment by fasmat** [comment]
Wow that was a simpler fix than I thought it would be 😅. CI is now looking green.

**Comment by STRRL** [comment]
> Wow that was a simpler fix than I thought it would be sweat_smile. CI is now looking green.

It seems just a temporary bypass solution that disables running test suites in parallel. 🙈 It just works and does not make some performance degression too much. I think we should let parallel back in the future.

**Review by STRRL — APPROVED** [review]
LGTM! Thanks a lot! 

🎉

**Comment by STRRL** [comment]
tichi bot does not respond to our command, I am going to merge this PR manually.

**Pull request merged**